File: yep/src/main/java/catchla/yep/loader/GithubUserInfoLoader.java
package catchla.yep.loader;

import android.accounts.Account;
import android.content.Context;
import android.support.annotation.NonNull;

import catchla.yep.model.GithubUserInfo;
import catchla.yep.util.YepAPI;
import catchla.yep.model.YepException;

/**
 * Created by mariotaku on 15/6/3.
 */
public class GithubUserInfoLoader extends CachedYepObjectLoader<GithubUserInfo> {

    private final String mYepUserId;

    public GithubUserInfoLoader(Context context, Account account, String yepUserId, boolean readCache, boolean writeCache) {
        super(context, account, GithubUserInfo.class, readCache, writeCache);
        mYepUserId = yepUserId;
    }

    @NonNull
    @Override
    protected String getCacheFileName() {
        return "cached_github_user_info_" + getAccount().name;
    }

    @Override
    protected GithubUserInfo requestData(final YepAPI yep, GithubUserInfo oldData) throws YepException {
        return yep.getGithubUserInfo(mYepUserId);
    }

}

File: src/main/kotlin/catchla/yep/loader/GithubUserInfoLoader.kt
package catchla.yep.loader

import android.accounts.Account
import android.content.Context
import catchla.yep.model.GithubUserInfo
import catchla.yep.model.YepException
import catchla.yep.util.YepAPI

/**
 * Created by mariotaku on 15/6/3.
 */
class GithubUserInfoLoader(
        context: Context,
        account: Account,
        private val yepUserId: String,
        readCache: Boolean,
        writeCache: Boolean
) : CachedYepObjectLoader<GithubUserInfo>(context, account, GithubUserInfo::class.java, readCache, writeCache) {

    override val cacheFileName: String
        get() = "cached_github_user_info_${account.name}"

    @Throws(YepException::class)
    override fun requestData(yep: YepAPI, oldData: GithubUserInfo?): GithubUserInfo {
        return yep.getGithubUserInfo(yepUserId)
    }

}
